lib/smart_proxy_dynflow/callback.rb

```diff
@@ -4,37 +4,37 @@ module Proxy
   class Dynflow
     module Callback
       class Request < Proxy::HttpRequest::ForemanRequest
-        def callback(callback, data)
-          payload = { :callback => callback, :data => data }.to_json
+        def callback(payload)
           response = send_request(request_factory.create_post('foreman_tasks/api/tasks/callback', payload))
           if response.code != "200"
             raise "Failed performing callback to Foreman server: #{response.code} #{response.body}"
           end
           response
         end
 
-        def self.send_to_foreman_tasks(callback, data)
-          self.new.callback(callback, data)
+        def self.send_to_foreman_tasks(payload)
+          self.new.callback(payload)
         end
       end
 
-      class Action < ::Dynflow::Action
-        def plan(callback, data)
-          plan_self(:callback => callback, :data => data)
+      class Core < Proxy::HttpRequest::ForemanRequest
+        def uri
+          @uri ||= URI.parse Proxy::Dynflow::Plugin.settings.core_url
         end
 
-        def run
-          Callback::Request.send_to_foreman_tasks(input[:callback], input[:data])
+        def relay(request, from, to)
+          path = request.env['REQUEST_PATH'].gsub(from, to)
```

Comment:
`request.env['REQUEST_PATH']` returns '/' on my setup, `request.path` seems to work